src/overlays/actors/ovl_Bg_Spot06_Objects/z_bg_spot06_objects.c

```diff
@@ -29,39 +64,440 @@ const ActorInit Bg_Spot06_Objects_InitVars = {
     (ActorFunc)BgSpot06Objects_Update,
     (ActorFunc)BgSpot06Objects_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot06_Objects/BgSpot06Objects_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot06_Objects/BgSpot06Objects_Destroy.s")
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000080, 0x00, 0x00 }, 0x00, 0x05, 0x01 },
+        { 1, { { 0, 0, -160 }, 18 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x39, 0x20, COLSHAPE_JNTSPH },
+    ARRAY_COUNT(sJntSphItemsInit),
+    sJntSphItemsInit,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static InitChainEntry sInitChainWaterPlane[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+void BgSpot06Objects_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot06Objects* this = THIS;
+    s32 pad;
+    CollisionHeader* colHeader = NULL;
+
+    this->switchFlag = thisx->params & 0xFF;
+    thisx->params = (thisx->params >> 8) & 0xFF;
+
+    osSyncPrintf("spot06 obj nthisx->arg_data=[%d]", thisx->params);
+
+    switch (thisx->params) {
+        case LHO_WATER_TEMPLE_ENTRACE_GATE:
+            Actor_ProcessInitChain(thisx, sInitChain);
+            DynaPolyActor_Init(thisx, DPM_UNK);
+            CollisionHeader_GetVirtual(&D_06000EE8, &colHeader);
+            this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+
+            if (LINK_IS_ADULT && (Flags_GetSwitch(globalCtx, this->switchFlag))) {
```

Comment:
It's pretty minor but imo there is no need for extra parens around conditions that don't have any operators (relevant in the rest of the file)
```suggestion
            if (LINK_IS_ADULT && Flags_GetSwitch(globalCtx, this->switchFlag)) {
```

Reply:
...oh I was told that everything needs parentheses now, which personally I think makes it much harder to read